var/spack/repos/builtin/packages/py-bglibpy/package.py

```diff
@@ -13,13 +13,11 @@ class PyBglibpy(PythonPackage):
     git = "ssh://bbpcode.epfl.ch/sim/BGLibPy"
 
     version('develop', branch='master')
-    version('4.1.2', commit='d01c9359edf92e39be5b840b5bb975da1c960988')
```

Comment:
I remember @matz-e saying in a PR not to remove the older versions.

Reply:
Yeah, but I looked on BB5. And all these old version show up in 'module av', so it's basically polluting that listing

Reply:
They will keep showing up in `module av` anyways. We don't delete old modules.

The spec/version that is in `bbp-packages.yaml` determines which new modules will be generated, and is the correct way to fix this issue.

Some cleanup can be nice, but just keeping one version around seems a little excessive to me. I.e., if you want to see if a change in output is relating to dependencies or a new version of a package, it's nice to be able to build two different versions with the same dependency tree. Also allows to do easy comparisons of performance, where the only variability is in the version number, and not the dependency tree.

Reply:
Then probably the solution would be that you only show the latest version in the modules.
We have to clean this, otherwise we could end up with a list of 50 bglibpy versions by the end of the year.

I've pinned the version in bbp-packages for now. It's a bit unfortunate we're gonna have to change this two files all the time, but anyway, at some point I'll probably anyway automate the creation of spack PRs when a new package of bglibpy is released.

Up until now I don't know of anybody who managed to use py-bglibpy using spack. So at least for now I don't think it hurts to clean up the versions and keep the package file clean.

Reply:
It should be fairly simple to generate ones own modules using Spack, but that may just be my inherent bias.

Old modules drop out of scope once we rotate the module directories at the beginning of a new month, so the list of modules will not grow indefinite.

Reply:
Ah ok
"Old modules drop out of scope once we rotate the module directories at the beginning of a new month"
That's a piece of the puzzle I was missing ;-)

Reply:
Yes, I realized that this hasn't been documented well :/